Sources/Private/CoreAnimation/Layers/TextLayer.swift

```diff
@@ -23,6 +23,13 @@ final class TextLayer: BaseCompositionLayer {
     fatalError("init(coder:) has not been implemented")
   }
 
+  override func setupAnimations(context: LayerAnimationContext) throws {
+    try super.setupAnimations(context: context)
+    let customText = context.textProvider.textFor(keypathName: context.currentKeypath.fullPath, sourceText: "text layer text")
```

Comment:
Would it be possible to pass in the actual source text like here: 
https://github.com/airbnb/lottie-ios/blob/167812e8ce59698144714b81a4132d3d188f2a39/Sources/Private/MainThread/LayerContainers/CompLayers/TextCompositionLayer.swift#L115

```suggestion
    let sourceText = try textLayerModel.text.exactlyOneKeyframe(context: context, description: "text layer text")
    let customText = context.textProvider.textFor(keypathName: context.currentKeypath.fullPath, sourceText: sourceText )
```

I think the `textLayerModel.text.exactlyOneKeyframe` code (currently on line 51) can be removed from `configureRenderLayer`.

Reply:
Sure, I think it's possible. 
Done in [8cb4ccd](https://github.com/airbnb/lottie-ios/pull/1723/commits/8cb4ccdfb8beb000563c0cdd1185c2dda08c119f) ✅ 

> I think the textLayerModel.text.exactlyOneKeyframe code (currently on line 51) can be removed from configureRenderLayer.

But I am not sure about that. Since we use the text later on in the `configureRenderLayer(:)`

```swift
    renderLayer.alignment = text.justification.textAlignment
    renderLayer.lineHeight = CGFloat(text.lineHeight)
    renderLayer.tracking = (CGFloat(text.fontSize) * CGFloat(text.tracking)) / 1000

    renderLayer.fillColor = text.fillColorData?.cgColorValue
    renderLayer.strokeColor = text.strokeColorData?.cgColorValue
    renderLayer.strokeWidth = CGFloat(text.strokeWidth ?? 0)
    renderLayer.strokeOnTop = text.strokeOverFill ?? false

    renderLayer.preferredSize = text.textFrameSize?.sizeValue
```

Reply:
Ah, haven't expanded far enough downwards, nevermind.